### DISCUSSION TITLE
Export labeled frames stuck after adding an SLP file as a video

### DISCUSSION THREAD

**Discussion by hl2199** [talmolab/sleap#1738] [Help!]
# Bug description

Export labeled frames stuck at 76.5% and does not progress. I suspect that this is due to my erroneous addition of a predictions.slp file as a video into the project. This also broke the videos pane, which now displays 0 videos instead of the total of 6. This means I cannot remove the incorrectly added video. I cannot replace this video either as the "replace video" options looks for another file with the same .slp extension. 

Please advise on how to resolve this issue and proceed with labels export. Thanks!  

## Screenshots

<img width="1914" alt="Screenshot 2024-03-30 at 9 28 41 PM" src="https://github.com/talmolab/sleap/assets/98616619/3d615674-3e32-45b2-b066-dbf50c89c2ae">

<img width="1690" alt="Screenshot 2024-03-30 at 9 32 35 PM" src="https://github.com/talmolab/sleap/assets/98616619/1e6859fd-21e3-4faf-ba8c-2472a018333f">

**Comment by Marielamend** [comment]
Hi @hl8536!
Just wanted to check in to see if you resolved this issue or needed additional help?

Best,
Mariela

**Comment by hl2199** [comment]
> Hi @hl8536! Just wanted to check in to see if you resolved this issue or needed additional help?
> 
> Best, Mariela

Fixed the issue by directly editing the project.slp file itself and with some patchwork with various files. However, this was not a straightforward fix and required going beyond what is possible within the intended user interface. It might be good for developers to address this issue for future users who may encounter something similar, as I could resolve this only with the help of someone who had significantly more experience working with SLEAP backend.

**Comment by talmo** [comment]
Hi @hl8536,

Thanks for letting us know. It sounds like the source of the issue is that you were able to add an SLP file as a video in the first place.

I'm reopening this so we can look into why this is even possible and to fix it. If you're able to share any details about how you fixed it, that might also help us reproduce it.

Alternatively, if you still have the SLP file (`male_broken.slp`) and [wouldn't mind sharing it with us](https://forms.gle/KebBPdjoHw2WMKmV8), it'll help us diagnose the root problem and how to prevent it from happening in the future.

Thanks,

Talmo

**Comment by hl2199** [comment]
> Hi @hl8536,
> 
> Thanks for letting us know. It sounds like the source of the issue is that you were able to add an SLP file as a video in the first place.
> 
> I'm reopening this so we can look into why this is even possible and to fix it. If you're able to share any details about how you fixed it, that might also help us reproduce it.
> 
> Alternatively, if you still have the SLP file (`male_broken.slp`) and [wouldn't mind sharing it with us](https://forms.gle/KebBPdjoHw2WMKmV8), it'll help us diagnose the root problem and how to prevent it from happening in the future.
> 
> Thanks,
> 
> Talmo

Submitted. Thanks for taking a look.

**Comment by talmo** [comment]
**Update:** Tracked down the root of the problem -- we can import SLP files as videos because we support packaged videos with embedded images. Specifically, here:

https://github.com/talmolab/sleap/blob/d4ad3bb37e5b76b027db62ba7f6926ea0d482e15/sleap/gui/dialogs/importvideos.py#L79-L97

https://github.com/talmolab/sleap/blob/d4ad3bb37e5b76b027db62ba7f6926ea0d482e15/sleap/io/video.py#L1628-L1640

https://github.com/talmolab/sleap/blob/d4ad3bb37e5b76b027db62ba7f6926ea0d482e15/sleap/io/video.py#L66

We could remove `"slp"` from that list, but I'm not sure if this will break other stuff. Maybe it's safer to just check for embedded files?

Alternatively, since we really don't support direct SLP file import as videos alone, maybe we _should_ just remove it from the import videos command.

In any case, the appropriate way of importing SLP files is via the File --> Merge functionality, which properly imports all the contents of a SLP file and deduplicates things where possible.

Given the low chance of this affecting others, I'm going to convert it to a discussion. If anyone else runs into this edge case, please let us know and we'll prioritize a smarter GUI functionality to prevent this from reoccurring.
